tests/integration/metrics_tests/test_metrics.py

```diff
@@ -38,11 +38,10 @@ async def get_address(ops_test: OpsTest, app_name=DATABASE_APP_NAME, unit_num=0)
 
 async def verify_endpoints(ops_test: OpsTest, app_name=DATABASE_APP_NAME):
     """Verifies mongodb endpoint is functional on a given unit."""
-    http = urllib3.PoolManager()
-
     for unit_id in range(len(ops_test.model.applications[app_name].units)):
```

Comment:
I believe we can get away with a session object, perhaps:
```suggestion
    with requests.Session() as http:
        for unit_id in range(len(ops_test.model.applications[app_name].units)):
```

Reply:
Actually if possible I would like to keep it as is (with my new changes from my recent commit)

my reasoning is that I want to keep tests between k8s and VM the same.

If this is a necessary change please let me know and I will make it on both charms

Reply:
I would say that we should perhaps add this to the VM charm as well (not urgent)